viewer/viewer-awt/src/main/java/org/icepdf/ri/common/SwingController.java

```diff
@@ -3553,6 +3563,103 @@ protected String generateNewSaveName(String fileName) {
         return null;
     }
 
+    /**
```

Comment:
This is really interesting work.   I would like to suggest mailing logic gets pushed out to a .util class similar to the BareBonesBrowserLauncher in util.  The Swing controller is already massive and this new block of functionality could be carved out into is own util.  

I'm still trying to get my head around the potential security issues of reading the registry if any.  In the new util class it might be helpful that some of the platform specific paths and params be made configurable even if only by system property.  Just thinking about servicing other organizations more easily.  

I'll do some testing on my windows virtual box.   I have noticed on my linux box the sendMailMailto fires up chrome and gmail but the url attachment didn't seem to get picked up.  I'll look into this further though and get you some more info.

Reply:
I moved the logic in the MailSender utility class.     
You're talking about setting a property like `mail.client` with e.g. `thunderbird` or `outlook`, etc ?      
Yeah I'm assuming some mail clients don't pick up the attachment parameter when parsing the mailto (given that it's not in the official syntax), the only way for these to work is to implement custom logic (like with Thunderbird), if it's even doable (I assume if the mail client is a web browser, it's simply not the case)